src/Gui/NavigationAnimation.cpp

```diff
@@ -0,0 +1,167 @@
+    prevTranslation = SbVec3f(0, 0, 0);
+
+    // Find an axis and angle to rotate from the camera orientation to the target orientation using post-multiplication
+    SbVec3f rotationAxisPost;
+    float angle;
+    SbRotation(navigation->getCamera()->orientation.getValue().inverse() * targetOrientation).getValue(rotationAxisPost, angle);
+    if (angle > M_PI) {
+        angle -= 2 * M_PI;
+    }
+
+    // Convert post-multiplication axis to a pre-multiplication axis
+    navigation->getCamera()->orientation.getValue().inverse().multVec(rotationAxisPost, rotationAxis);
+
+    angularVelocity = angle / duration();
+    linearVelocity = targetTranslation / duration();
+}
+
+/**
+ * @param value The elapsed time
+ */
+void FixedTimeAnimation::update(const QVariant& value)
+{
+    SoCamera* camera = navigation->getCamera();
+    if (!camera) {
+        return;
+    }
+
+    float angle = value.toFloat() * angularVelocity;
+    SbVec3f translation = value.toFloat() * linearVelocity;
+
+    SbRotation rotation(rotationAxis, angle - prevAngle);
+
+    camera->position = camera->position.getValue() - prevTranslation;
+    navigation->reorientCamera(camera, rotation, rotationCenter);
+    camera->position = camera->position.getValue() + translation;
+
+    prevAngle = angle;
+    prevTranslation = translation;
+}
+
+/**
+ * @param finished True when the animation is finished, false when interrupted
+ */
+void FixedTimeAnimation::onStop(bool finished)
+{
+    if (finished) {
+        SoCamera* camera = navigation->getCamera();
+        if (!camera) {
+            return;
+        }
+
+        // Set exact target orientation
+        camera->orientation = targetOrientation;
+        camera->position = camera->position.getValue() + targetTranslation - prevTranslation;
+    }
+}
+
+/**
+ * @param navigation The navigation style
+ * @param axis The rotation axis in screen coordinates
+ * @param velocity The angular velocity in radians per second
+ */
+SpinningAnimation::SpinningAnimation(NavigationStyle* navigation, const SbVec3f& axis,
+                                     float velocity)
+    : NavigationAnimation(navigation)
+    , rotationAxis(axis)
+{
+    setDuration((2 * M_PI / velocity) * 1000.0);
+    setStartValue(0.0);
+    setEndValue(2 * M_PI);
+    setLoopCount(-1);
+}
+
+void SpinningAnimation::initialize()
+{
+    prevAngle = 0;
+
+    navigation->setViewing(true);
+    navigation->setViewingMode(NavigationStyle::SPINNING);
+}
+
+/**
+ * @param value The angle in radians
+ */
+void SpinningAnimation::update(const QVariant& value)
+{
+    SoCamera* camera = navigation->getCamera();
+    if (!camera) {
+        return;
+    }
+
+    SbRotation deltaRotation = SbRotation(rotationAxis, value.toFloat() - prevAngle);
+    navigation->reorientCamera(camera, deltaRotation);
+
+    prevAngle = value.toFloat();
+}
+
+/**
+ * @param finished True when the animation is finished, false when interrupted
+ */
+void SpinningAnimation::onStop(bool finished)
```

Comment:
`finished` isn't used. Either use `Q_UNUSED(finished)` or declare it as `[[maybe_unused]]`

Reply:
Resolved in 2fec778d31a44790854e6ea224d0a436664d8096